src/sources/soundsource.cpp

```diff
@@ -52,6 +53,18 @@ QString SoundSource::getTypeFromFile(const QFileInfo& fileInfo) {
         // https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/mimetype.20sometimes.20wrong
         return fileSuffix;
     }
+    if (normalisedFilename.endsWith(QLatin1String(".stem.mp4"))) {
+        // STEM suffix extends the ".mp4" suffix. This is because stem files are
+        // meant to fall back to MP4 decoder, in case a player doesn't have the
+        // capability to extract the different stem track. However, we need to
+        // detect that early as the file natively looks like a traditional MP4
+        // with multiple tracks (audio and video)
+        return QLatin1String("stem.mp4");
+    }
+    if (normalisedFilename.endsWith(QLatin1String(".stem.m4a"))) {
+        // STEM suffix may also extends the ".m4a" suffix for the same reason as above
+        return QLatin1String("stem.m4a");
+    }
```

Comment:
We normally prioritize the mime type over the file extension. I think mp4 and m4a both have "audio/mp4".
So I think we need to look up the mime type first and than check if the file name is "*stem.*". This will finally allow any random file extension which has a stem file name. Or we may be more strict and limit it to "stem.m4a" and "stem.mp4".
Optimal would be a magic number to look up. But probably not worth the effort.

Reply:
At which offset are the atoms located? Is the lookup fast? Maybe we can do this.

Reply:
Cool, than let's do it. 
Is it correct to do it for all audio/mp4 mime types?

Reply:
Sounds good to me.